packages/block-editor/src/components/rich-text/use-enter.js

```diff
@@ -67,7 +68,16 @@ export function useEnter( props ) {
 
 			const { text, start, end } = _value;
 
-			if ( event.shiftKey ) {
+			if ( shouldDisableEditing ) {
+				// Simulate the cursor is at the end of the rich text.
+				_value.start = _value.text?.length;
+				_value.end = _value.text?.length;
+				splitValue( {
```

Comment:
Would it be enough to only set the position at the end of the block and let the logic do its thing?

Reply:
Just setting the `end` seems to work fine, but I thought it was more reliable to set both. Let me try to explain why. From what I understood:

* start: defines the start of the cursor.
* end: defines the end of the cursor.

So, if I select multiple characters they are different. While if I don't select any character, they are the same:

![Screenshot 2024-02-23 at 12 58 51](https://github.com/WordPress/gutenberg/assets/34552881/7992d713-ed2e-4de2-9f30-7bd8f16b3756)

In this case `start` is 2 and end is 4.

![Screenshot 2024-02-23 at 13 00 00](https://github.com/WordPress/gutenberg/assets/34552881/3a132645-fb12-4ad7-b5f8-197924b61694)

In this other case, both `start` and `end` are 2.

When the rich text is not editable, both `start` and `end` are undefined. So, to simulate placing the cursor in the last character I thought that was the way to go.

Any thoughts?

Reply:
I was talking about different types of changes. At the moment, the code exits early when `shouldDisableEditing` is set to `true`, it runs `splitValue`. My question was whether all the forthcoming conditions would be able to handle it instead.

Reply:
Right, adding a new line in the text doesn't make too much sense here 👍🏻 

Everything else seems to work pretty well so this fix should be good in general.